android/modules/media/src/java/ti/modules/titanium/media/TiCameraXActivity.java

```diff
@@ -0,0 +1,570 @@
+							String[] columns = {
+								MediaStore.Images.Media.DISPLAY_NAME,
+								MediaStore.Images.Media.DATA
+							};
+							String[] selectionArgs = { imageName };
+							String queryString = MediaStore.Images.ImageColumns.DISPLAY_NAME + " = ? ";
+							Cursor cursor = contentResolver.query(mediaUri, columns, queryString, selectionArgs, null);
+							if ((cursor != null) && cursor.moveToNext()) {
+								name = getStringFrom(cursor, 0);
+								path = getStringFrom(cursor, 1);
+							}
+
+							if (name != null && !name.equals("")) {
+								TiBlob blob = TiBlob.blobFromFile(TiFileFactory.createTitaniumFile(path, false));
+								KrollDict response = MediaModule.createDictForImage(blob, blob.getMimeType());
+								successCallback.callAsync(callbackContext, response);
+							}
+						} else {
+							// normal file
+							File file = new File(outputUri.getPath());
+							TiBlob blob = TiBlob.blobFromFile(TiFileFactory.createTitaniumFile(file.getPath(), false));
+							KrollDict response = MediaModule.createDictForImage(blob, blob.getMimeType());
+							successCallback.callAsync(callbackContext, response);
+						}
+
+						if (cameraActivity != null && autohide) {
+							cameraActivity.finish();
+						}
+					}
+				}
+
+				@Override
+				public void onError(ImageCaptureException error)
+				{
+					if (errorCallback == null) {
+						Log.e(TAG, error.toString());
+						return;
+					}
+
+					KrollDict dict = new KrollDict();
+					dict.putCodeAndMessage(error.getImageCaptureError(), error.toString());
+					dict.put(TiC.PROPERTY_MESSAGE, error.toString());
+					errorCallback.callAsync(callbackContext, dict);
+				}
+			});
+		} catch (Exception ex) {
+			KrollDict dict = new KrollDict();
+			dict.put(TiC.PROPERTY_MESSAGE, "Error taking a picture");
+			errorCallback.callAsync(callbackContext, dict);
+		}
+	}
+
+	@SuppressLint("MissingPermission")
+	public static void startVideoCapture()
+	{
+		KrollDict recordingDict = new KrollDict();
+
+		Consumer<VideoRecordEvent> vre = videoRecordEvent -> {
+			if (videoRecordEvent instanceof VideoRecordEvent.Start) {
+				// Handle the start of a new active recording
+			} else if (videoRecordEvent instanceof VideoRecordEvent.Pause) {
+				// Handle the case where the active recording is paused
+			} else if (videoRecordEvent instanceof VideoRecordEvent.Resume) {
+				// Handles the case where the active recording is resumed
+			} else if (videoRecordEvent instanceof VideoRecordEvent.Finalize) {
+				VideoRecordEvent.Finalize finalizeEvent =
+					(VideoRecordEvent.Finalize) videoRecordEvent;
+				// Handles a finalize event for the active recording, checking Finalize.getError()
+				int error = finalizeEvent.getError();
+				if (error != VideoRecordEvent.Finalize.ERROR_NONE) {
+					// error
+				}
+				Uri uri = finalizeEvent.getOutputResults().getOutputUri();
+
+				if (uri.toString().startsWith("content://")) {
+					String path = "";
```

Comment:
This looks like duplicate logic from above at [L111](https://github.com/tidev/titanium_mobile/pull/12395/files#diff-52633e11fe636615f24f4c067fc7d0dadd18e950519f41d4b159905616aaee87R135)

Reply:
kind of. Just with `MediaStore.Video`